### DISCUSSION TITLE
Is there a way to load a db file directly?

### DISCUSSION THREAD

**Discussion by xelephas** [duckdb/duckdb-wasm#1780] [Q&A] [answered]
I would like to instantiate a duckdb directly from a database file, is that possible? 

So something like this:

```
duckdb.connect("file.db")
```

And the `file.db` could be uploaded by the user, and read using [FileReader](https://developer.mozilla.org/en-US/docs/Web/API/FileReader)

**Comment by severo** [comment, marked as answer]
Maybe with ATTACH:

https://duckdb.org/docs/sql/statements/attach
https://duckdb.org/docs/guides/network_cloud_storage/duckdb_over_https_or_s3

https://github.com/search?q=repo%3Aduckdb%2Fduckdb-wasm+attach&type=code
